src/unix.rs

```diff
@@ -17,17 +17,19 @@ use std::thread::{self, Builder, JoinHandle};
 use std::time::Duration;
 
 #[derive(Debug)]
-pub enum Client {
-    /// `--jobserver-auth=R,W`
-    Pipe { read: File, write: File },
-    /// `--jobserver-auth=fifo:PATH`
-    Fifo {
-        file: File,
-        path: PathBuf,
-        /// it can only go from false -> true but not the other way around, since that
-        ///  could cause a race condition.
-        is_non_blocking: AtomicBool,
-    },
+enum ClientCreationArg {
+    Fds { read: c_int, write: c_int },
+    Fifo(Box<Path>),
+}
+
+#[derive(Debug)]
+pub struct Client {
+    read: File,
+    write: File,
+    creation_arg: ClientCreationArg,
+    /// it can only go from Some(false) -> Some(true) but not the other way around, since that
+    ///  could cause a race condition.
+    is_non_blocking: Option<AtomicBool>,
```

Comment:
We could have another

```suggestion
    non_blocking_read: Option<(File, AtomicBool)>,
```

which should improve performance of `acquire`, after `try_acquire` is called, while also enable the fifo optimization for jobserver created by `Client::mk`.

But given how rare `try_acquire` is used, I suspect that is unnecessary, and with #90 it won't be needed on newer Linux kernel anyway.